src/OrchardCore/OrchardCore.Abstractions/Json/Dynamic/JsonDynamicBase.cs

```diff
@@ -0,0 +1,13 @@
+using System.Dynamic;
+using System.Text.Json.Nodes;
+
+#nullable enable
+
+namespace System.Text.Json.Dynamic;
+
+public abstract class JsonDynamicBase : DynamicObject
+{
+    public abstract JsonNode? Node { get; }
```

Comment:
Why `Node` and not `Value`, would you use the member name `Value` to determine the target type?

Can we use something like this to check the target type in this PR?

```C#
 case JsonValueKind.String:
                if (memberName == "Value")
                {
                    if (currentNodeValue.TryGetValue<DateTime>(out var datetime))
                    {
                        result = datetime;
                        return true;
                    }

                    if (currentNodeValue.TryGetValue<TimeSpan>(out var timeSpan))
                    {
                        result = timeSpan;
                        return true;
                    }
                }

                result = currentNodeValue.GetString();
                return true;
```

Reply:
It would be a `Value` property on `JsonDynamicValue` only, not on the base class. This is the base class for `JsonDynamicObject` and `JsonDynamicArray` too, so `Value` would be misleading. 

But getting the target type like this is always only a guess, I'm not sure if we should add that back.

Reply:
ok ，I've tested it. It's also read as a string in 1.8, no problem. 
```js
const content = contentItem("4s7hn4k9rf0yfzez8ds7mrs9d2");
return {
    data:{
        year:new Date(content.Content.Test.DateField.Value).getFullYear()
    }
}
```